Purchases/FoundationExtensions/DateFormatter+Extensions.swift

```diff
@@ -14,32 +14,62 @@
 
 import Foundation
 
-extension DateFormatter {
+/// A type that can convert from and to `Dates`.
+protocol DateFormatterType {
 
-    static let iso8601SecondsDateFormatter: DateFormatter = {
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
-        return dateFormatter
+    func string(from date: Date) -> String
+    func date(from string: String) -> Date?
+
+}
+
+extension DateFormatter: DateFormatterType {}
+extension ISO8601DateFormatter: DateFormatterType {}
+
+internal extension ISO8601DateFormatter {
+
+    private static let withMilliseconds: DateFormatterType = {
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [
+            .withInternetDateTime,
+            .withFractionalSeconds
+        ]
+
+        return formatter
     }()
 
-    static func date(fromISO8601SecondsOrMillisecondsString maybeDateString: String?) -> Date? {
-        return (Self.iso8601SecondsDateFormatter.date(fromString: maybeDateString)
-            ?? Self.iso8601MilliSecondsDateFormatter.date(fromString: maybeDateString))
-    }
+    private static let noMilliseconds: DateFormatterType = {
```

Comment:
We try to keep all of the [private methods in a single extension](https://github.com/RevenueCat/purchases-ios/blob/main/SwiftStyleGuide.swift#L106), and preferably after the more public and internal ones, so that when you first start reading a file, you see all of the higher-level stuff first. That way you can understand the contents at a glance and only need to keep reading if you want to delve deeper into the implementation

Reply:
Yup sorry! I'll fix this.